pkg/data/domain/catalog/service.go

```diff
@@ -39,68 +39,72 @@ func New(config Config) (Interface, error) {
 	return s, nil
 }
 
-// Get fetches a list of appcatalog CRs optionally filtered by name.
+// Get fetches a list of catalog CRs optionally filtered by name.
 func (s *Service) Get(ctx context.Context, options GetOptions) (Resource, error) {
 	var resource Resource
 	var err error
 
 	if len(options.Name) > 0 {
-		resource, err = s.getByName(ctx, options.Name, options.LabelSelector)
+		resource, err = s.getByName(ctx, options.Namespace, options.Name, options.LabelSelector)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 
 		return resource, nil
 	}
 
-	resource, err = s.getAll(ctx, options.LabelSelector)
+	resource, err = s.getAll(ctx, options)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
 	return resource, nil
 }
 
-func (s *Service) getAll(ctx context.Context, labelSelector labels.Selector) (Resource, error) {
-	appCatalogCollection := &Collection{}
+func (s *Service) getAll(ctx context.Context, options GetOptions) (Resource, error) {
+	catalogCollection := &Collection{}
 	var err error
 
 	{
-		lo := &runtimeclient.ListOptions{
-			LabelSelector: labelSelector,
-		}
-
-		appCatalogs := &applicationv1alpha1.AppCatalogList{}
+		catalogs := &applicationv1alpha1.CatalogList{}
 		{
-			err = s.client.K8sClient.CtrlClient().List(ctx, appCatalogs, lo)
+			err = s.client.K8sClient.CtrlClient().List(ctx, catalogs, runtimeclient.InNamespace(options.Namespace))
 			if apimeta.IsNoMatchError(err) {
 				return nil, microerror.Mask(noMatchError)
 			} else if err != nil {
 				return nil, microerror.Mask(err)
-			} else if len(appCatalogs.Items) == 0 {
+			} else if len(catalogs.Items) == 0 {
 				return nil, microerror.Mask(noResourcesError)
 			}
 		}
 
-		for _, appCatalog := range appCatalogs.Items {
-			a := AppCatalog{
-				CR: omitManagedFields(appCatalog.DeepCopy()),
+		for _, catalog := range catalogs.Items {
+			// We hide catalog CRs from the giantswarm namespace by
+			// default as these are internal.
```

Comment:
If -A is true we hide internal catalogs in the giantswarm namespace.

You can still use -n giantswarm if you have access.